### PR TITLE
feat(util): support IP range/CIDR filtering in event logs

### PR DESCRIPTION
**Pull request by ayushi-work**
## Changes

Adds IP range/CIDR matching support for audit/event log filtering, addressing #3480.

### New files
- **`pkg/util/ip.go`** — `IPMatcher` utility supporting CIDR notation (`192.168.1.0/24`), IP ranges (`192.168.1.1-192.168.1.255`), wildcard patterns (`192.168.*.*`), and single exact IPs (backward compatible)
- **`pkg/util/ip_test.go`** — Comprehensive tests covering IPv4, IPv6, edge cases

### Modified files
- **`service/admin/list.go`** — Added `IPMatcherFromCondition()` and `IPMatcherFromSearch()` convenience methods on `AdminListService`

### CODE REVIEW

File: service/admin/list.go [hunk 1]
```diff
@@ -17,6 +18,30 @@ type (
 	AdminListServiceParamsCtx struct{}
 )
 
+// IPMatcherFromCondition returns an IPMatcher for the given condition key.
+// If the condition value is empty, returns nil (no filter).
+// The condition value can be a single IP, CIDR notation, IP range, or wildcard pattern.
+// Returns an error only if the condition value is non-empty but invalid.
+func (s *AdminListService) IPMatcherFromCondition(key string) (*util.IPMatcher, error) {
```
Comment (HFO4) [inline service/admin/list.go]: These 2 methods are never used.

**Review by HFO4 — CHANGES_REQUESTED** [review]

**Review by ayushi-work — COMMENTED** [review]
I have removed the unused helper methods from `AdminListService`. The PR now only contains the self-contained `IPMatcher` utility in `pkg/util/` which can be imported by audit log handlers when needed.

**Comment by ayushi-work** [comment]
Hi @HFO4 bumping this up!
